### PR TITLE
feat: [PLG-2434]: Added API endpoint for retrieving a Stripe client secret

### PR DESCRIPTION
**Pull request by thomas-grasman**
## Describe your changes
Added and endpoint for generating and returning a client secret. Client secrets are required for displaying Stripe's credit card element. A setup intent is created using Stripe's API and the setup intent will contain the secret we need.

Also, added the isDefault field to the creditCardDTO for displaying the customer's active credit card on the billing page.

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/48688)

### CODE REVIEW

**Comment by thomas-grasman** [comment]
trigger uts

File: 935-subscription-sdk/src/main/java/io/harness/subscription/helpers/impl/StripeHandlerImpl.java [hunk 1]
```diff
@@ -280,4 +281,14 @@ private void sendTelemetryEvent(String event, String email, String accountId, St
     telemetryReporter.sendTrackEvent(event, email, accountId, properties,
         ImmutableMap.<Destination, Boolean>builder().put(Destination.AMPLITUDE, true).build(), SUBSCRIPTION);
   }
+
+  public SetupIntent retrieveSetupIntent(String customerId) {
+    try {
+      Map<String, Object> params = new HashMap<>();
+
```
Comment (Xin-Harness) [inline 935-subscription-sdk/src/main/java/io/harness/subscription/helpers/impl/StripeHandlerImpl.java:288]: Would like to get a better understanding of adding an empty map to the retrieve setup intent method

File: 930-ng-license-manager/src/main/java/io/harness/subscription/resource/SubscriptionResource.java [hunk 2]
```diff
@@ -354,6 +354,24 @@ public class SubscriptionResource {
   //    return ResponseDTO.newResponse(subscriptionService.listStripeCustomers(accountIdentifier));
   //  }
 
+  @GET
+  @Path("/client_secret")
```
Comment (Xin-Harness) [inline 930-ng-license-manager/src/main/java/io/harness/subscription/resource/SubscriptionResource.java:358]: better to use hyphen then underscore in API

**Review by Xin-Harness — CHANGES_REQUESTED** [review]

**Review by neelam-harness — CHANGES_REQUESTED** [review]

**Review by thomas-grasman — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/48688)* status: 7 file(s) to review. 2 change requests. 7. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`930-ng-license-manager/src/main/java/io/harness/subscription/resource/SubscriptionResource.java` line 358 at r1](https://reviewable.io/reviews/harness/harness-core/48688#gh-1222178727:-NXRy_0G2eiQqeGa0vv3:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/962906967818cf1a47536dd2daeb495d9ee073e3/930-ng-license-manager/src/main/java/io/harness/subscription/resource/SubscriptionResource.java#L358)):*
<details><summary><i>Previously, Xin-Harness (Xin Shao) wrote…</i></summary><blockquote>

better to use hyphen then underscore in API
</blockquote></details>

Done.
___
*[`930-ng-license-manager/src/main/java/io/harness/subscription/resource/SubscriptionResource.java` line 360 at r1](https://reviewable.io/reviews/harness/harness-core/48688#gh-1223541909:-NXRy_TY725J4LGhiKBS:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/962906967818cf1a47536dd2daeb495d9ee073e3/930-ng-license-manager/src/main/java/io/harness/subscription/resource/SubscriptionResource.java#L360)):*
<details><summary><i>Previously, neelam-harness wrote…</i></summary><blockquote>

The api is named `retrieveSetupIntent`, the summary states that it `"Creates a client secret for the Stripe customer"`. I think let's make the description, operation and the api name all aligned. 
</blockquote></details>

Done.
___
*[`930-ng-license-manager/src/main/java/io/harness/subscription/services/SubscriptionService.java` line 47 at r1](https://reviewable.io/reviews/harness/harness-core/48688#gh-1223543069:-NXRyi_dCTNLmXHr67Su:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/962906967818cf1a47536dd2daeb495d9ee073e3/930-ng-license-manager/src/main/java/io/harness/subscription/services/SubscriptionService.java#L47)):*
<details><summary><i>Previously, neelam-harness wrote…</i></summary><blockquote>

add a javadoc for the method
</blockquote></details>

Done.
___
*[`930-ng-license-manager/src/main/java/io/harness/subscription/services/impl/SubscriptionServiceImpl.java` line 565 at r1](https://reviewable.io/reviews/harness/harness-core/48688#gh-1223542453:-NXS-0wc2XaW7DdPtphp:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/962906967818cf1a47536dd2daeb495d9ee073e3/930-ng-license-manager/src/main/java/io/harness/subscription/services/impl/SubscriptionServiceImpl.java#L565)):*
<details><summary><i>Previously, neelam-harness wrote…</i></summary><blockquote>

please remove the commented code
</blockquote></details>

Done.
___
*[`935-subscription-sdk/src/main/java/io/harness/subscription/helpers/impl/StripeHandlerImpl.java` line 288 at r1](https://reviewable.io/reviews/harness/harness-core/48688#gh-1222177071:-NXS-3a67t5y8EDL0SUi:bdew31t) ([raw file](https://github.com/harness/harness-core/blob/962906967818cf1a47536dd2daeb495d9ee073e3/935-subscription-sdk/src/main/java/io/harness/subscription/helpers/impl/StripeHandlerImpl.java#L288)):*
<details><summary><i>Previously, Xin-Harness (Xin Shao) wrote…</i></summary><blockquote>

Would like to get a better understanding of adding an empty map to the retrieve setup intent method
</blockquote></details>

I need to declare an object here in order to choose one of the overloaded methods. Since I need any specific options changed for this setup intent, I just initialized an empty object. What I should have done is just declared null.
___
*[`950-subscription-beans/src/main/java/io/harness/subscription/dto/CardDTO.java` line 14 at r1](https://reviewable.io/reviews/harness/harness-core/48688#gh-1223548752:-NXS-RR00CEly2wMfBbG:b3jqn8u) ([raw file](https://github.com/harness/harness-core/blob/962906967818cf1a47536dd2daeb495d9ee073e3/950-subscription-beans/src/main/java/io/harness/subscription/dto/CardDTO.java#L14)):*
<details><summary><i>Previously, neelam-harness wrote…</i></summary><blockquote>

i think this should be `java.util.Optional`
</blockquote></details>

java.util.Optional is for internal nullable objects. I need the annotation Optional to define the data contract.
___
*[`950-subscription-beans/src/main/java/io/harness/subscription/dto/CardDTO.java` line 35 at r1](https://reviewable.io/reviews/harness/harness-core/48688#gh-1223548087:-NXS-lM91X19RSx7Z-av:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/962906967818cf1a47536dd2daeb495d9ee073e3/950-subscription-beans/src/main/java/io/harness/subscription/dto/CardDTO.java#L35)):*
<details><summary><i>Previously, neelam-harness wrote…</i></summary><blockquote>

can rename it `isDefaultCard` ?
</blockquote></details>

Done.

**Comment by thomas-grasman** [comment]
trigger uts

**Comment by thomas-grasman** [comment]
trigger uts

**Review by kapil-harness — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/48688)* status: 7 file(s) to review. 3 change requests. 7. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`930-ng-license-manager/src/main/java/io/harness/subscription/resource/SubscriptionResource.java` line 360 at r1](https://reviewable.io/reviews/harness/harness-core/48688#gh-1223541909:-NXTvKHjAAaEC7kuyXd4:b-h0c63y) ([raw file](https://github.com/harness/harness-core/blob/962906967818cf1a47536dd2daeb495d9ee073e3/930-ng-license-manager/src/main/java/io/harness/subscription/resource/SubscriptionResource.java#L360)):*
<details><summary><i>Previously, thomas-grasman (Thomas Grasman) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

how a @GET API is creating a client secret? should it not be a @POST API?
___
*[`930-ng-license-manager/src/main/java/io/harness/subscription/resource/SubscriptionResource.java` line 364 at r4](https://reviewable.io/reviews/harness/harness-core/48688#-NXTvXlmBYyRPCBwIEcC:-NXTvXlmBYyRPCBwIEcD:b-nzhi3s) ([raw file](https://github.com/harness/harness-core/blob/3aec060ffd7ae377bfbcb988878e0d6e5a2a64d7/930-ng-license-manager/src/main/java/io/harness/subscription/resource/SubscriptionResource.java#L364)):*
> ```Java
>       {
>         @io.swagger.v3.oas.annotations.responses.
>         ApiResponse(responseCode = "default", description = "Returns a Stripe credit card element client secret")
> ```

here it says that it just returns a client secret and not create it
___
*[`930-ng-license-manager/src/main/java/io/harness/subscription/services/SubscriptionService.java` line 47 at r1](https://reviewable.io/reviews/harness/harness-core/48688#gh-1223543069:-NXTvo7w9v_KFBPalx3v:bj7gl9b) ([raw file](https://github.com/harness/harness-core/blob/962906967818cf1a47536dd2daeb495d9ee073e3/930-ng-license-manager/src/main/java/io/harness/subscription/services/SubscriptionService.java#L47)):*
<details><summary><i>Previously, thomas-grasman (Thomas Grasman) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

we should remove the comments

**Review by Xin-Harness — APPROVED** [review]

**Review by thomas-grasman — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/48688)* status: 7 file(s) to review. 2 change requests. 8. unresolved discussions. 1 of 2 approvals obtained. 

___
*[`930-ng-license-manager/src/main/java/io/harness/subscription/resource/SubscriptionResource.java` line 360 at r1](https://reviewable.io/reviews/harness/harness-core/48688#gh-1223541909:-NXVY-MO9Ke8Y0R3aMi1:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/962906967818cf1a47536dd2daeb495d9ee073e3/930-ng-license-manager/src/main/java/io/harness/subscription/resource/SubscriptionResource.java#L360)):*
<details><summary><i>Previously, kapil-harness (Kapil Choudhary) wrote…</i></summary><blockquote>

how a @GET API is creating a client secret? should it not be a @POST API?
</blockquote></details>

Done.
___
*[`930-ng-license-manager/src/main/java/io/harness/subscription/resource/SubscriptionResource.java` line 364 at r4](https://reviewable.io/reviews/harness/harness-core/48688#-NXTvXlmBYyRPCBwIEcC:-NXVY4HP0a2YknPQXHnz:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/3aec060ffd7ae377bfbcb988878e0d6e5a2a64d7/930-ng-license-manager/src/main/java/io/harness/subscription/resource/SubscriptionResource.java#L364)):*
<details><summary><i>Previously, kapil-harness (Kapil Choudhary) wrote…</i></summary><blockquote>

here it says that it just returns a client secret and not create it
</blockquote></details>

Done.
___
*[`930-ng-license-manager/src/main/java/io/harness/subscription/services/SubscriptionService.java` line 47 at r1](https://reviewable.io/reviews/harness/harness-core/48688#gh-1223543069:-NXVY5IE5BXnv3hmtk2e:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/962906967818cf1a47536dd2daeb495d9ee073e3/930-ng-license-manager/src/main/java/io/harness/subscription/services/SubscriptionService.java#L47)):*
<details><summary><i>Previously, kapil-harness (Kapil Choudhary) wrote…</i></summary><blockquote>

we should remove the comments
</blockquote></details>

Done.

**Review by kapil-harness — APPROVED** [review]
> Reviewed 1 of 4 files at r3, 6 of 6 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/48688)* status: 7 file(s) to review. 1 change requests. 7. unresolved discussions.

**Comment by thomas-grasman** [comment]
trigger review

**Review by neelam-harness — APPROVED** [review]

**Pull request merged**